build.sbt

```diff
@@ -159,7 +159,7 @@ lazy val dockerSettings = Seq(
   Docker / packageName := "adopttapir",
   dockerUsername := Some("softwaremill"),
   dockerUpdateLatest := true,
-  Docker / publishLocal := (Docker / publishLocal).dependsOn(copyWebapp).value,
+  Docker / publish := (Docker / publish).dependsOn(copyWebapp).value,
```

Comment:
I've taken a look at the `Docker / publish` dependencies, and it seems it still depends on `publishLocal`:
`[info] Dependencies:
[info]  backend / Docker / dockerExecCommand
[info]  backend / Docker / dockerAliases
[info]  backend / Docker / dockerBuildxPlatforms
[info]  backend / Docker / publishLocal
[info]  backend / Docker / dockerBuildOptions
[info]  backend / Docker / publish / streams
[info]  backend / Docker / stage
`
So additional `dependOn` probably won't work. I would move it a little higher, so f.e. compile java or build jar tasks should depend on the `copyWebapp`.